test/conformance/api/v1/util.go

```diff
@@ -121,28 +123,24 @@ func validateDataPlane(t pkgTest.TLegacy, clients *test.Clients, names test.Reso
 // Validates the state of Configuration, Revision, and Route objects for a runLatest Service.
 // The checks in this method should be able to be performed at any point in a
 // runLatest Service's lifecycle so long as the service is in a "Ready" state.
-func validateControlPlane(t *testing.T, clients *test.Clients, names test.ResourceNames, expectedGeneration string, imagesForMultipleContainers ...map[string]string) error {
+func validateControlPlane(t *testing.T, clients *test.Clients, names test.ResourceNames, expectedGeneration string) error {
 	t.Log("Checking to ensure Revision is in desired state with", "generation", expectedGeneration)
 	err := v1test.CheckRevisionState(clients.ServingClient, names.Revision, func(r *v1.Revision) (bool, error) {
 		if ready, err := v1test.IsRevisionReady(r); !ready {
 			return false, fmt.Errorf("revision %s did not become ready to serve traffic: %w", names.Revision, err)
 		}
-		// for multi containers there will be more than one container and names.Image is a string,
-		// so in order to validate imageDigest for each images validateControlPlane accepting optional field called imagesForMultipleContainers
-		if names.Image == "" {
-			for _, value := range imagesForMultipleContainers {
-				for _, v := range r.Status.ContainerStatuses {
-					if image, ok := value[v.Name]; ok {
-						if validDigest, err := shared.ValidateImageDigest(t, image, v.ImageDigest); !validDigest {
-							return false, fmt.Errorf("imageDigest %s is not valid for imageName %s: %w", v.ImageDigest, image, err)
-						}
+		for _, v := range r.Status.ContainerStatuses {
+			// For multi container scenario images and digest coming from respective entities will not be in same sequence,
+			// So this change ensures correctness of image w.r.t digest.
+			// It works for both single and multi container scenario
+			f := regexp.MustCompile(`\@+`).Split(v.ImageDigest, -1)
+			for _, image := range names.Images {
+				if image == strings.Split(f[0], "/")[2] {
```

Comment:
this seems a bit odd, how do we know the image is the third bit? Also why do we need to compare the image at all rather than the containerStatus.Name? (assuming we can't just rely on order being maintained in the resolution)

Reply:
modified :+1: